File: c++/hierarchical inheritance account.cpp
#include<iostream>
using namespace std;
class account
{
	public:
		int actno;
		int bal;
		int interest;
		void getdata(int a,int b,int c)
		{
			actno=a;
			bal=b;
			interest=c;
			
		}
};
class saving:public account
{
	public:
		int deposite(int amt)
		{
			bal=bal+interest+amt;
			return bal;
		}
};
class current:public account
{
	public:
		int deposite(int amt)
		{
			bal=bal+amt;
			return bal;
		}
};
int main()
{
	saving s1;
	current c1;
	int result,actno,bal,interest,amount;
	cout<<"enter actno,bal,int,amt:";
	cin>>actno>>bal>>interest>>amount;
	s1.getdata(actno,bal,interest);
	result=s1.deposite(amount);
	cout<<"saving account"<<result<<endl;
	c1.getdata(actno,bal,interest);
	result=c1.deposite(amount);
	cout<<"current amount"<<result;
	return 0;
	
}

File: opps/hierarchical_inheritance_account.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace hierarchical__inheritance_amount
{
    class account
    {
       public int balance = 1000;
       public int actno;
        public void deposit(int amount)
        {
            Console.WriteLine("this is deposit method of account class");

        }
    }
    class saving : account
    {
        public void deposit(int amount)
        {
            int interest = 500;
            balance = balance + amount + interest;
            Console.WriteLine("amount deposited with interest bal is:" + balance);
        }
    }
    class current:account
    {
        public void deposit(int amount)
        {
            balance = balance + amount;
            Console.WriteLine("amount deposited without interest bal is: " + balance);
        }
    }

    internal class Program
    {
        static void Main(string[] args)
        {
            saving s = new saving();
            s.actno = 111;
            s.deposit(1000);

            current c = new current();
            c.actno = 121;
            c.deposit(1000);
            Console.ReadKey();

        }
    }
}
